db/blip_handler.go

```diff
@@ -931,7 +931,23 @@ func (bh *blipHandler) handleProveAttachment(rq *blip.Message) error {
 		return base.HTTPErrorf(http.StatusBadRequest, "no digest sent with proveAttachment")
 	}
 
-	attData, err := bh.db.GetAttachment(AttachmentKey(digest))
+	docID := ""
+	attachmentAllowedKey := digest
+	if bh.blipContext.ActiveSubprotocol() == BlipCBMobileReplicationV3 {
+		docID = rq.Properties[ProveAttachmentID]
+		if docID == "" {
+			return base.HTTPErrorf(http.StatusBadRequest, "no docID sent with proveAttachment")
+		}
+		attachmentAllowedKey = docID + digest
+	}
+
+	allowedAttachment := bh.allowedAttachment(attachmentAllowedKey)
+	if allowedAttachment.counter <= 0 {
+		return base.HTTPErrorf(http.StatusForbidden, "Attachment's doc not being synced")
+	}
```

Comment:
This looks like new handling that we weren't previously doing. Is it a gap in our `proveAttachment` implementation? What scenario is this check trying to catch?

Reply:
Yeah, this change is not required, proveAttachment handling can be refactored instead.